### PR TITLE
Update diffblue.yml and add CYOA fallback phases

### PR DESCRIPTION
**Pull request by majakusber**
The changes were done as follows. 

1. Given:
- diffblue.yml file used on Symphony, Reladomo, MyCollab and E-commerce Demo (they use the same options in the three phases),
- default platform options specified here https://github.com/diffblue/platform/blob/develop/docker/application-server/src/config/tools.json,
I constructed the first three phases to use the same options as the above standard benchmarks (previously I used the options from Tika) and for better readability, I explicitly state any options that differ from the default ones in any phase (so would get overridden in any phase).

2. Using the choose-your-own-adventure feature, I added two fallback phases - in case of timeouts or insufficient coverage (only in the third phase), fall back to a phase with load-containing-class-only and subsequently single-function-only.

Testing on stream-lib benchmark:
- old config: platform predicted 55.31% and verified 48.15% (in 27min),
- new config: platform predicted 66.69% (+11.38%) and verified 56.69% (+8.54%) (in 36min)

If approved, the diffblue.yml will be used on all 20 new benchmarks. Note that the purpose of this change is not to tweak the file as much as possible for all options but rather to quickly gain coverage using some low hanging fruit.

### CODE REVIEW

File: diffblue.yml [hunk 1]
```diff
@@ -5,20 +5,61 @@ phases:
  timeout: 300
  cbmcArguments:
    classpath: '/tools/cbmc/models-simple-overlay.jar:/tools/cbmc/models.jar:.'
-   max-nondet-array-length: 100
-   java-max-vla-length: false
+   java-assume-inputs-non-null: true
+   java-max-vla-length: 48
+   max-nondet-array-length: 10
+   max-nondet-string-length: 10
+   string-printable: true
```
Comment (allredj) [inline diffblue.yml:12]: :bulb: Maybe keep that to false? `string-printable` can't improve coverage

**Review by allredj — APPROVED** [review]
:owl:

File: diffblue.yml (hunk 1 above)
Comment (antlechner) [reply diffblue.yml:12]: I think this is set to `true` in phase 1 on a lot of benchmarks because in phase 1 we're aiming for pretty tests? Not objecting to changing it, just thinking we may want to check with whoever made the decision to set it to `true` here.
I'm guessing the same applies to `java-assume-inputs-non-null`?

File: diffblue.yml [hunk 2]
```diff
@@ -5,20 +5,61 @@ phases:
  timeout: 300
```
Comment (antlechner) [inline diffblue.yml:4]: :grey_question: I can't comment on line 2, but @allredj didn't you say that you've seen cases where a high depth caused problems? Should we set it to lower values in phases 1 and 2? Or maybe that should be left for the fine-tuning and is out of scope of this PR.

File: diffblue.yml [hunk 3]
```diff
@@ -5,20 +5,61 @@ phases:
...
 -
  timeout: 300
  cbmcArguments:
    classpath: '/tools/cbmc/models.jar:.'
-   max-nondet-array-length: 100
-   java-max-vla-length: false
+   java-assume-inputs-non-null: false
+   java-max-vla-length:  192
+   max-nondet-array-length: 30
+   max-nondet-string-length: 10
```
Comment (antlechner) [inline diffblue.yml]: :question: I know this is how it appears in current `diffblue.yml` files but I think this is a mistake. The value shouldn't be lower than in phase 2. Maybe set it to 100? I think it's 100 somewhere in phase 3...

File: diffblue.yml [hunk 4]
```diff
@@ -5,20 +5,61 @@ phases:
...
+   string-printable: false
+   throw-runtime-exceptions: true
    unwind: 3
+-
+ timeout: 300
```
Comment (antlechner) [inline diffblue.yml:45]: :grey_question: Would it be worth splitting this one into two phases, one with lower values and `throw-runtime-exceptions: false` and a second one with higher values and `throw-runtime-exceptions: true`? Assuming that many executions will get 100% coverage in the first run this might save us some time. Same for the last phase.

**Review by antlechner — COMMENTED** [review]
One :question: comment that I think should be addressed, just ideas and nitpicks otherwise.

File: diffblue.yml (hunk 2 above)
Comment (majakusber) [reply diffblue.yml:4]: We seem to use `depth: false` on all our benchmarks at the moment, the platform default values are 1500, 3000 and 6000. I think this is an option to be fine-tuned later as it's not clear how much say an extra 1000 depth would affect things in a general case.

File: diffblue.yml (hunk 1 above)
Comment (majakusber) [reply diffblue.yml:12]: Yes, both these options seem to be for pretty tests. They should not really decrease coverage though as far as I understand (`java-assume-inputs-non-null` is only used in the first phase, so the missing tests can still use null later on).

File: diffblue.yml (hunk 3 above)
Comment (majakusber) [reply diffblue.yml]: Yes, it's 100 in the platform default options. Will change it to 100.

File: diffblue.yml (hunk 2 above)
Comment (allredj) [reply diffblue.yml:4]: `depth 1500` works in 95% of cases. I would use it in most phases. There are cases where we need higher values, like 3000. Maybe we could use that in fallback phases.

I've seen at least a few cases where `depth: false` is bad and prevents test-gen from terminating.

File: diffblue.yml (hunk 4 above)
Comment (majakusber) [reply diffblue.yml:45]: Hm, maybe.. I'll try and run the benchmark with the setup where as fallback we would have copy of first phase with `load-containing-class-only: true`, then copy of third phase with `load-containing-class-only: true`, and then the same for `single-function-only: true`. I'm curious if and how much more time it will take.. and coverage of course.

File: diffblue.yml (hunk 2 above)
Comment (majakusber) [reply diffblue.yml:4]: Ok, I'll use 1500 in the first two phases, 3000 in third phase. I'm testing fallbacks that copy 1st and 3rd phase now so that means the higher depth would appear there too.

File: diffblue.yml (hunk 4 above)
Comment (antlechner) [reply diffblue.yml:45]: That sounds like a good idea, let me know what the result was and when I should re-review this PR. I'm happy with anything beyond the current discussions on this PR to be left for fine-tuning later. :slightly_smiling_face:

**Comment by majakusber** [comment]
I pushed a commit with the additional changes as suggested and ran the benchmark, the numbers are even better (although it takes a bit longer):
    * platform predicted 70.65% (+15.34%) and verified 61.44% (+13.29%) (in 43min)
